src/array_buffer.rs

```diff
@@ -526,6 +538,53 @@ impl ArrayBuffer {
     }
   }
 
+  /// Returns a new standalone BackingStore backed by an object that dereferences
+  /// to a mutable slice of bytes. The object is dereferenced once, and the resulting slice's
```

Comment:
I don't understand this. It seems that this does the following:
* Copy the object that holds the bytes to the heap and then:
* Wrap it into a BackingStore.
If you want to do that, why don't you just create a BackingStore from the underlying bytes?

Reply:
I think it might make more sense to have a generic implementation for any type that can create a `BackingStore` for anything that implements `Into<Box<[u8]>>` which covers (I believe) `Vec<u8>`, `String`, `[u8]`. That's more useful because those `Into` implementations generally try to avoid copying (definitely for `Vec`s and `String`s).

Reply:
In this case we're boxing something (assuming that it allocates bytes somewhere else on the heap) and then asking it to deref from there. The challenge is with BytesMut, common in the tokio ecosystem, which doesn't have `Into<Box<[u8]>>`. It _does_ have conversion to Vec<u8> so maybe there's another option.

To be honest I'm rethinking this PR and I might tweak it a bit if I can make it work more generically.